### PR TITLE
Endpoint byos check

### PR DESCRIPTION
**Pull request by jesusbv**
## Description

Add new endpoint to check billing code from metadata

## Change Type

*Please select the correct option.*

- [ ] **Bug Fix** (a non-breaking change which fixes an issue)
- [X] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [X] I have verified that my code follows RMT's coding standards with `rubocop`.
- [X] I have reviewed my own code and believe that it's ready for an external review.
- [X] I have provided comments for any hard-to-understand code.
- [ ] I have documented the `MANUAL.md` file with any changes to the user experience.
- X ] RMT's test coverage remains at 100%.
- [ ] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

## Other Notes

Please use this space to provide notes or thoughts to the team, such as tips on how to review/demo your changes.

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/307973/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged e80a40b6400fb0385bd2945ac63c6a0fe24dc4b4 into 314791ad30771a91a2ebb12e96a8461e7f364bd1)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/307974/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 98b7d197ac2b58163334a6f908efc8ad450bcf10 into 314791ad30771a91a2ebb12e96a8461e7f364bd1)

**Comment by jesusbv** [comment]
Probably will move the test to its own suite instead of being part of `products_controller_spec`

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/308086/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 3e3364396b42985d471cd01086ac507a9f645e9e into 314791ad30771a91a2ebb12e96a8461e7f364bd1)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/308088/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 346472e28b599f84d416a78826d6b8b05dfae571 into 314791ad30771a91a2ebb12e96a8461e7f364bd1)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/308098/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 8666fadf0e0b98cc753b1852fd11d1207cec393b into 314791ad30771a91a2ebb12e96a8461e7f364bd1)

File: engines/instance_verification/app/controllers/instance_verification/billing_check_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module InstanceVerification
+  class BillingCheckController < ::ApplicationController
```
Comment (digitaltom) [inline engines/instance_verification/app/controllers/instance_verification/billing_check_controller.rb:2]: Why this nested path, and not `engines/instance_verification/app/controllers/instance_verification_controller.rb`?

**Review by digitaltom — CHANGES_REQUESTED** [review]
Please move/drop the route into the engine. 
Other than that, LGTM, but I obviously cannot really test it, so it would make sense if you test that in the cloud.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/309416/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 706fe5d8dba86080051ab970671d6400c3d87fef into dc30add40c117b7545c8bb992175cefbce5ea0fe)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/309427/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d838807f06e459c98bb5f030a13a904eabd46d1f into dc30add40c117b7545c8bb992175cefbce5ea0fe)

**Review by rjschwei — APPROVED** [review]

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/310769/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d838807f06e459c98bb5f030a13a904eabd46d1f into 40bb4866a081f45874c9e7a40caa838c96c5d19f)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/311625/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d838807f06e459c98bb5f030a13a904eabd46d1f into 2e2a9361ec65f59f4cec75bc493b6fc215e26ce9)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/312121/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 8b5d786256a7db1ff94dd2bdc8dcb67f049c513f into 24ec6046bbc0c856dfc78f07d98f9d2615134113)

**Pull request merged**